### PR TITLE
Dont pop up Next Up Tooltip during live playback

### PR DESCRIPTION
**Pull request by johnBartos**
- Detach 'change:position' handler when in Live mode.
- Reattach handler on a new Next Up item (will get detached again if it's live)

JW7-3105

### CODE REVIEW

File: src/js/view/components/nextuptooltip.js [hunk 1]
```diff
@@ -157,6 +158,7 @@ define([
             var streamType = this._model.get('streamType');
             if (streamType === 'LIVE' || streamType === 'DVR') {
                 model.off('change:duration', this.onDuration, this);
+                model.off('change:position', this.onElapsed, this);
```
Comment (robwalch) [inline src/js/view/components/nextuptooltip.js]: Live/DVR/VOD state can change when duration changes. Maybe these listeners should be removed and added on 'change:streamType'?

File: src/js/view/components/nextuptooltip.js [hunk 2]
```diff
@@ -144,9 +144,10 @@ define([
             }
 
             this._nextButton.toggle(true);
-            // Listen for duration changes to determine the offset
-            // for when next up should be shown
+            // Listen for duration changes to determine the offset from the end for when next up should be shown
             this._model.on('change:duration', this.onDuration, this);
+            // Listen for position changes so we can show the tooltip when the offset has been crossed
+            this._model.on('change:position', this.onElapsed, this);
```
Comment (robwalch) [inline src/js/view/components/nextuptooltip.js]: You remove the duration handler above, but not position.... you could be adding multiple position handlers.

Comment (johnBartos) [inline src/js/view/components/nextuptooltip.js]: Now detaching every time on`streamType` change and only adding when it's `VOD`

File: src/js/view/components/nextuptooltip.js (hunk 1 above)
Comment (johnBartos) [inline src/js/view/components/nextuptooltip.js]: Done

File: src/js/view/components/nextuptooltip.js [hunk 3]
```diff
@@ -50,15 +49,15 @@ define([
...
-            var style = {};
             this.nextUpImage = new Image();
             this.nextUpImage.onload = (function() {
                 this.nextUpImage.onload = null;
             }).bind(this);
             this.nextUpImage.src = url;
 
-            style.backgroundImage = 'url("' + url + '")';
-            return style;
+            return {
```
Comment (johnBartos) [inline src/js/view/components/nextuptooltip.js]: Cleanup

**Pull request merged**